docs/source/more_types.rst

```diff
@@ -307,6 +307,27 @@ redundancy, it's possible to replace default values in overload definitions with
     def get_model(model_or_pk: int | M, flag: bool = True) -> M | None:
         ...
 
+The overload that applies is chosen depending on the type of the ``model_or_pk``
+argument. On the other hand, the return type in the following example depends on
+an argument with a default value. To ensure that the overloads do not overlap,
+a default value is specified for only one of them.
+
+.. code-block:: python
+
+    from typing import overload, Literal
+
+    # This overload variant applies when "rounded" is omitted or set to False.
+    @overload
+    def pi(rounded: Literal[False] = ...) -> float: ...
```

Comment:
This is not correct. This function will always be matched. Without a default should be the first one.

Reply:
I’m happy to change this, but at least in this case, both overloads are apparently matched. I tested with this file:
```python
from typing import overload, Literal

@overload
def pi(rounded: Literal[False] = ...) -> float: ...

@overload
def pi(rounded: Literal[True]) -> int: ...

def pi(rounded=False):
    return 3 if rounded else 3.14159

reveal_type(pi(rounded=False))
reveal_type(pi())
reveal_type(pi(rounded=True))
```
mypy 0.982 tells me:
```
default.py:12: note: Revealed type is "builtins.float"
default.py:13: note: Revealed type is "builtins.float"
default.py:14: note: Revealed type is "builtins.int"
```
Swapping the order of the overloads doesn’t change the output. There’s also no warning about shadowing, wouldn’t that be printed otherwise?

Reply:
I made the change.